test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/BooleanJSTest.scala

```diff
@@ -7,32 +7,16 @@
 \*                                                                      */
 package org.scalajs.testsuite.compiler
 
-import org.scalajs.jasminetest.JasmineTest
-import scala.scalajs.js
-
-object ShortTest extends JasmineTest {
-
-  describe("Short primitives") {
-
-    it("should always be in their range") {
-      def test(x: Int, y: Short): Unit =
-        expect(x.toShort).toEqual(y)
+import org.junit.Test
+import org.junit.Assert._
 
-      test(0, 0)
-      test(-500, -500)
-      test(-90000, -24464)
-      test(123456789, -13035)
-      test(-40000, 25536)
-      test(65536, 0)
-      test(32768, -32768)
-
-      def testC(x: Char, y: Short): Unit =
-        expect(x.toShort).toEqual(y)
-
-      testC(-1.toChar, -1)
-      testC(200.toChar, 200)
-      testC(60000.toChar, -5536)
-    }
+import scala.scalajs.js
 
+class BooleanJSTest {
+  @Test
+  def `primitive_operations_on_booleans_should_return_boolean`(): Unit = {
+    assertEquals(js.typeOf(true & false), "boolean")
```

Comment:
Lol, these tests are completely useless: they're constant-folded *by scalac*. We're not at all testing those operations are the IR level, nor, a fortiori, at the JS level -_-'

We should fix that as a follow up PR (let's not mix a change in behavior like that in the middle of a purely refactoring PR)